### DISCUSSION TITLE
Splashscreen - Cordova 12.0.0

### DISCUSSION THREAD

**Discussion by leonelgregioauler** [apache/cordova#412] [General]
Hello everyone !

I have a doubt, I'm upgrading to Cordova 12, is the first time that I'm using splashscreen without plugin. 
Before, my application had used the 10 version.

I don't know how is the code that I have to do in config.xml and splashscreen.xml

Anyone can help me, please ?

Att,

**Comment by breautek** [comment]
Reading the SplashScreen docs would be a start: https://cordova.apache.org/docs/en/11.x/core/features/splashscreen/index.html

For Android, at minimum you'll need to use [AndroidWindowSplashScreenAnimatedIcon](https://cordova.apache.org/docs/en/11.x/core/features/splashscreen/index.html#androidwindowsplashscreenanimatedicon). It can be a PNG if desired. Note that this is an icon, not a fullscreen image. See the [Android Docs](https://developer.android.com/develop/ui/views/launch/splash-screen) for more information on the elements/mechanics of the Android splashscreen. Cordova implements the Icon with background strategy, so you're icon should be 240x240dp in resolution (960x960px using a PNG, which is 4x the size of the base resolution for `xxxhdpi` support).

For iOS, they use launch screens which can be supplied by a [single 2732x2732px image](https://cordova.apache.org/docs/en/11.x/core/features/splashscreen/index.html#single-image-launch-screen).

> I don't know how is the code that I have to do in config.xml and splashscreen.xml

There is no `splashscreen.xml` (unless if that's your android drawable vector). The splashscreen is controlled entirely by `config.xml`.

**Reply by leonelgregioauler** [reply]
So, in the documentation it asks for the splashscreen.xml path. So I include the icon file path, instead ?

![image](https://github.com/apache/cordova/assets/12154804/b751d365-0b31-4635-b5d2-2ebeb698a504)

**Reply by breautek** [reply]
Yes, the path is relative to the cordova project. It's a file you supply.

So the value as shown in the example, `"res/screen/android/splashscreen.xml"` means cordova will look for the file at `<cordova-project-root>/res/screen/android/splashscreen.xml`

The file can either be a PNG, or an AVD (Android Vector Drawable). SVGs may work as well but mileage will vary. Android will attempt to convert SVGs into AVDs, but the android build system doesn't implement the full SVG spec and there are many SVG features that the build system will not work with.

If using a PNG, you should x4 the suggested dimensions which are in density-independent pixels, which is where the 960x960px resolution comes from. If you're using a vector like SVG or AVD, then the resolution can be the same as the suggested density-independent pixel since the files themselves can be scaled to the appropriate resolutions.

**Reply by leonelgregioauler** [reply]
Thank you for now
I will still implement it, then I'll tell you to close the thread

**Reply by leonelgregioauler** [reply]
Hello !

I implemented the functionality and got success! This implementation is better than the old one, certainly

Thank you !


![image](https://github.com/apache/cordova/assets/12154804/858146e0-47ee-41d4-ad45-155870326fcf)
